setup/src/installer.ts

```diff
@@ -155,13 +155,21 @@ export async function installTool(
       }
       await ghcup(tool, version, os);
       if (await isInstalled(tool, version, os)) return;
+      if (tool === 'ghc') {
+        await ghcupGHCrelease(version, os);
+        if (await isInstalled(tool, version, os)) return;
+      }
       await apt(tool, version);
       break;
     case 'win32':
       await choco(tool, version);
       break;
     case 'darwin':
       await ghcup(tool, version, os);
+      if (tool === 'ghc') {
```

Comment:
We're missing a `if (await isInstalled(tool, version, os)) return;` right here because we don't want to try and download GHC unless `ghcup` failed.